samples/DispatcherSample/HttpMethodEndpointSelector.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Dispatcher;
+
+namespace DispatcherSample
+{
+    public class HttpMethodEndpointSelector : EndpointSelector
```

Comment:
This is an off the cuff implementation from memory of MVC's http method selection + backtracking